py_zipkin/zipkin.py

```diff
@@ -169,7 +170,17 @@ def __init__(
     def __call__(self, f):
         @functools.wraps(f)
         def decorated(*args, **kwargs):
-            with self:
+            with zipkin_span(
+                service_name=self.service_name,
+                span_name=self.span_name,
+                zipkin_attrs=self.zipkin_attrs,
```

Comment:
Should we be disallowing certain passed-in arguments in this form? In my mind, the decorator form doesn't make any sense for zipkin_attrs or annotations, as these should change on a per-call basis. You're probably never going to want to log span after span with, say, the same trace_id, or the same timestamped annotations.

We could check for `self.zipkin_attrs` and raise an exception if a `zipkin_span` with these set is used in a decorator fashion.

Just a thought.